stream/src/public_key/connection.rs

```diff
@@ -664,10 +664,9 @@ mod tests {
             // Spawn a mock peer that responds with a listener response from wrong peer
             context.with_label("mock_peer").spawn({
                 move |mut context| async move {
-                    use chacha20poly1305::KeyInit;
-
                     // Read the hello from dialer
                     let msg = recv_frame(&mut peer_stream, 1024).await.unwrap();
+                    let _ = handshake::Hello::<PublicKey>::decode(msg.clone()).unwrap();
```

Comment:
While we could avoid this clone, I find it much clearer to follow (and will be easier to debug a test failure) if it stays next to msg.

Reply:
Makes sense!